azure-iot-device/azure/iot/device/iothub/pipeline/method_invoke_pipeline.py

```diff
@@ -0,0 +1,71 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import sys
+from azure.iot.device.common.evented_callback import EventedCallback
+from azure.iot.device.common.pipeline import (
+    pipeline_stages_base,
+    pipeline_ops_base,
+    pipeline_stages_http,
+)
+from . import (
+    constant,
+    pipeline_stages_iothub,
+    pipeline_ops_iothub,
+    pipeline_ops_method_invoke,
+    pipeline_stages_method_invoke_http,
+)
+from azure.iot.device.iothub.auth.x509_authentication_provider import X509AuthenticationProvider
+
+logger = logging.getLogger(__name__)
+
+
+class MethodInvokePipeline(object):
+    """Pipeline to communicate with Edge.
+    Uses HTTP.
+    """
+
+    def __init__(self, auth_provider, pipeline_configuration):
+        self._pipeline = (
```

Comment:
I thought we wanted retry or something in this pipeline?

Reply:
Maybe, but I think we need to add those later.

---
In reply to: [351491603](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r351491603) [](ancestors = 351491603)

Reply:
agreed

---
In reply to: [352791072](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r352791072) [](ancestors = 352791072,351491603)